File: app/src/main/java/com/example/ruru/sophiesblog/java/extends_/abstractAndInterface/activity/BaseActivity.java
package com.example.ruru.sophiesblog.java.extends_.abstractAndInterface.activity;

import android.os.Bundle;

import com.example.ruru.sophiesblog.R;

import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getContentViewId());

        initMembersView(savedInstanceState);
    }

    public abstract int getContentViewId();

    public abstract void initMembersView(Bundle savedInstanceState);
}

File: app/src/main/java/com/example/ruru/sophiesblog/android/activity/base/viewModelTwo/base/BaseActivity.kt
package com.example.ruru.sophiesblog.android.activity.base.viewModelTwo.base

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.*
import com.example.ruru.sophiesblog.android.activity.base.viewModelTwo.BViewModel

open class BaseActivity: AppCompatActivity(), LifecycleOwner {

  fun <T: BaseViewModel> bindViewModel(cls: Class<T>): T {
    var viewModel = ViewModelProvider(this).get(cls)
    bindProgress(viewModel.progress)
    return viewModel
  }

  fun bindProgress(progressData: MutableLiveData<Boolean>) {
    progressData.observe(this, Observer {
      if (it) {
        showProgressDialog()
      } else {
        dismissProgressDialog()
      }
    })
  }

  private fun showProgressDialog() {

  }

  private fun dismissProgressDialog() {

  }
}
